### DISCUSSION TITLE
Implement captcha element

### DISCUSSION THREAD

**Discussion by adamberecz** [vueform/vueform#67] [Ideas]
Implement captcha element

**Comment by letmaik** [comment]
This is an implementation of Cloudflare's Turnstile captcha:

```vue

<script>
import VueTurnstile from 'vue-turnstile';
import { defineElement } from '@vueform/vueform'

export default defineElement({
    name: 'TurnstileElement',
    nullValue: '',
    props: {
      siteKey: {
        type: String,
        required: false,
        default: 'custom',
      }
    },
    setup(props, { element }) {
    },
    components: {
        VueTurnstile
    },
})

</script>

<template>
    <ElementLayout>
        <template #element>
            <vue-turnstile ref="input" :site-key="siteKey" v-model="value" />
        </template>

        <!-- Default element slots -->
        <template v-for="(component, slot) in elementSlots" #[slot]>
            <slot :name="slot" :el$="el$">
                <component :is="component" :el$="el$" />
            </slot>
        </template>
    </ElementLayout>
</template>
```

The element value has the token (or nothing if human validation failed) which then has to be validated server-side as well at form submission.

**Comment by adamberecz** [comment]
Released in `1.9.6`
https://vueform.com/reference/captcha-element

**Reply by letmaik** [reply]
Cool! Some feedback:
- https://vueform.com/reference/captcha-element#property-captcha-options says boolean type, seems incorrect?
- https://vueform.com/reference/captcha-element#property-value is this the token that is sent with the form and must be validated server-side?
- https://vueform.com/reference/captcha-element#option-provider typo "avaialable"

I think some more details on the flow would be helpful, otherwise folks may incorrectly implement it and not actually validate the captcha server-side.

**Reply by adamberecz** [reply]
Thanks for the suggestions! They all got addressed. The backend flow is not described in details, but a note has been added that backend validation is required: https://vueform.com/reference/captcha-element#basic-usage
